### PR TITLE
k8s 1.25

### PR DESCRIPTION
**Pull request by jenshu**
# Description

Made the following updates to support running Gloo Edge on Kubernetes 1.25:

## Build-bot:
- run with helm v3.11.2 and kubectl to v1.25.8 via [cloud-builders update](https://github.com/solo-io/cloud-builders/pull/105)
## GitHub Actions / regression tests:
- use `go-version-file` to get Go version from go.mod (not related to k8s 1.25 but nice to have)
- removed `actions/cache` since setup-go v4 [enables caching by default](https://github.com/actions/setup-go#v4)
- run regression tests on k8s 1.25
- for Istio tests, use Istio 1.17 which [supports k8s 1.23-1.26](https://istio.io/latest/docs/releases/supported-releases/)
- updated nightly test matrix for 1.14 to run on k8s 1.23 and 1.25, and made helm version configurable
## Library updates:
- upgraded skv2, helm, and k8s.io libraries to support k8s 1.25
- removed the [klog shim](https://github.com/solo-io/gloo/pull/1880) since newer versions of klog write to stderr by default
## Helm:
- changed the way we push helm charts to the OCI registry, due to [changes in helm 3.7](https://github.com/helm/helm/releases/tag/v3.7.0)
    - the new charts are located in a subdirectory `gcr.io/solo-public/gloo-helm/gloo` (used to be at the top level in `gcr.io/solo-public/gloo-helm`)
- Changed `PodDisruptionBudget` from `policy/v1beta1` to `policy/v1` per [deprecation guide](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25)

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/checks/checks_test.go [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-
-import (
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
-	"github.com/solo-io/gloo/pkg/utils/modfile"
-)
-
-var _ = Describe("Checks", func() {
-
-	It("should used forked klog instead of klog", func() {
```
Comment (nfuden) [inline test/checks/checks_test.go:11]: Are we sure this is safe? We still seem to have klog references in some of our deps. Given this dont we need to maintain this? We can check that for now we are safe by running with a  read only root file system deployment but im not sure that will prevent a regression

**Review by Rachael-Graham — DISMISSED** [review]
docs LGTM

**Review by nfuden — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

File: .github/workflows/regression-tests.yaml [hunk 2]
```diff
@@ -65,7 +65,7 @@ jobs:
       fail-fast: false
       matrix:
         kube-e2e-test-type: ['gateway', 'gloo', 'ingress', 'helm', 'gloomtls', 'glooctl', 'upgrade', 'istio']
-        kube-version: [{ node: 'v1.24.7@sha256:5c015142d9b60a0f6c45573f809957076514e38ec973565e2b2fe828b91597f5', kubectl: 'v1.24.7', kind: 'v0.17.0'}]
+        kube-version: [{ node: 'v1.25.3', kubectl: 'v1.25.8', kind: 'v0.17.0', helm: 'v3.11.2' }]
```
Comment (bewebi) [inline .github/workflows/regression-tests.yaml:68]: Same note about specifying a SHA

**Review by bewebi — COMMENTED** [review]
Looks good!

One question/comment

**Review by nfuden — APPROVED** [review]

**Review by bewebi — APPROVED** [review]

**Pull request merged**